crates/ide-assists/src/handlers/from_to_cfg_attr.rs

```diff
@@ -0,0 +1,184 @@
+        ))
+        .clone_for_update();
+        let attr_syntax = edit.make_syntax_mut(attr.syntax().clone());
+        ted::replace(attr_syntax, cfg_attr.syntax());
+
+        if let Some(snippet_cap) = ctx.config.snippet_cap {
+            if let Some(first_meta) =
+                cfg_attr.meta().and_then(|meta| meta.token_tree()).and_then(|tt| tt.l_paren_token())
+            {
+                edit.add_tabstop_after_token(snippet_cap, first_meta)
+            }
+        }
+    };
+    acc.add(
+        AssistId("from_to_cfg_attr", AssistKind::Refactor),
+        "Convert to `cfg_attr`",
+        range,
+        handle_source_change,
+    );
+    Some(())
+}
+fn from_cfg_attr(acc: &mut Assists, attr: ast::Attr) -> Option<()> {
+    let range = attr.syntax().text_range();
+    let meta = attr.meta()?;
+    let meta_tt = meta.token_tree()?;
+    let mut inner_attrs = Vec::with_capacity(1);
+    let mut found_comma = false;
+    let mut iter = meta_tt.token_trees_and_tokens().skip(1).peekable();
+    while let Some(tt) = iter.next() {
+        if let NodeOrToken::Token(token) = &tt {
+            if token.kind() == T![')'] {
+                break;
+            }
+            if token.kind() == T![,] {
+                found_comma = true;
+                continue;
+            }
+        }
+        if !found_comma {
+            continue;
+        }
+        let Some(attr_name) = tt.into_token().and_then(|token| {
+            if token.kind() == T![ident] {
+                Some(make::ext::ident_path(token.text()))
+            } else {
+                None
+            }
+        }) else {
+            continue;
+        };
+        let tt = iter.next()?;
+
+        let meta = match tt {
+            NodeOrToken::Node(tt) => make::meta_token_tree(attr_name, tt),
+            NodeOrToken::Token(token) if token.kind() == T![=] => {
+                let tt = iter.next()?;
+                if let Some(literal) = tt.into_token().and_then(|v| {
+                    if v.kind().is_literal() {
+                        Some(make::expr_literal(v.text()))
+                    } else {
+                        None
+                    }
+                }) {
+                    make::meta_expr(attr_name, ast::Expr::Literal(literal))
+                } else {
+                    return None;
+                }
+            }
+            NodeOrToken::Token(token) if token.kind() == T![,] => make::meta_path(attr_name),
+            _ => return None,
+        };
+
+        inner_attrs.push(make::attr_outer(meta));
+    }
+    let handle_source_change = |f: &mut SourceChangeBuilder| {
+        let inner_attrs = inner_attrs.iter().map(|it| it.to_string()).join("\n");
+        f.replace(range, inner_attrs);
+    };
+    acc.add(
+        AssistId("from_to_cfg_attr", AssistKind::Refactor),
+        "Extract Inner Attributes from `cfg_attr`",
+        range,
+        handle_source_change,
+    );
+    Some(())
+}
+#[cfg(test)]
+mod tests {
```

Comment:
Some test case ideas (includes what you already have)
- `#[derive$0(Debug)]`, `#[derive(Debug$0)]`, `#[derive$0(Debug, Clone)]`, `#[derive(Debug$0, Clone)]` (would be neat if for the last one it would split the debug derive out, not needed for this PR just a thought)
    These need the `//- minicore: derive` fixture prefix so that things resolve correctly
- `#[foo$0]`, `#[foo$0]`,  `#[cfg_attr(bar, foo$0)]`,  `#[cfg_attr(bar$0, foo)]`,  `#[cfg_attr$0(bar, foo)]`,  `#[cfg_attr$0(bar, foo, qux)]` (last one here needs to be turned into two attributes notably)
- `#![foo$0]`, `#![cfg_attr(bar, foo$0)]`

Reply:
> (would be neat if for the last one it would split the debug derive out, not needed for this PR just a thought)
These need the //- minicore: derive fixture prefix so that things resolve correctly

I started working on that and I started having issues with getting the snippet and new text to be in the right spot. I decided to stop fighting with it.  But yeah. I can add this in a future PR. As it would be helpful.